File: java/baiduads-sdk/src/test/com/baidu/dev2/api/sdk/account/GetAccountInfo.java
package com.baidu.dev2.api.sdk.account;

import com.baidu.dev2.api.sdk.account.api.AccountService;
import com.baidu.dev2.api.sdk.account.model.ApiAccountQueryRequest;
import com.baidu.dev2.api.sdk.account.model.GetAccountInfoRequestWrapper;
import com.baidu.dev2.api.sdk.account.model.GetAccountInfoResponseWrapper;
import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.invoke.ApiClient;
import com.baidu.dev2.api.sdk.invoke.Configuration;

import java.util.ArrayList;

/**
 * SDK指南 演示代码
 */
public class GetAccountInfo {

    public static void main(String[] args) {
        try {
            // 全局设置代理
            ApiClient defaultApiClient = Configuration.getDefaultApiClient();
            defaultApiClient.setBasePath("https://{proxyHost}:{proxyPort}");

            // init service
            AccountService accountService = new AccountService();

            // init requset header
            ApiRequestHeader requestHeader = new ApiRequestHeader();
            requestHeader.setToken("xxxxx");
            requestHeader.setUsername("xxxxx");
            requestHeader.setPassword("xxxxx");

            // init requset
            ApiAccountQueryRequest request = new ApiAccountQueryRequest();
            ArrayList<String> fields = new ArrayList<>();
            fields.add("pictureOptimizeSegmentStatus");
            fields.add("balance");
            fields.add("userLevel");
            request.setAccountFields(fields);

            // init requster wrapper
            GetAccountInfoRequestWrapper requestWrapper = new GetAccountInfoRequestWrapper();
            requestWrapper.setHeader(requestHeader);
            requestWrapper.setBody(request);

            GetAccountInfoResponseWrapper responseWrapper = accountService.getAccountInfo(requestWrapper);
            System.out.println(responseWrapper);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: python/baiduads-sdk/test/account/get_account_info.py
"""
SDK指南 演示代码
"""
from baiduads import Configuration, ApiClient
from baiduads.account.api import AccountService
from baiduads.account.model import ApiAccountQueryRequest, GetAccountInfoRequestWrapper
from baiduads.common.model import ApiRequestHeader


def main():
    # init service
    account_service = AccountService()

    # init request header
    request_header = ApiRequestHeader(
        token="xxxxx",
        username="xxxxx",
        password="xxxxx",
        _spec_property_naming=True
    )

    # init request
    account_query_request = ApiAccountQueryRequest(
        accountFields=["pictureOptimizeSegmentStatus", "balance", "userLevel"],
        _spec_property_naming=True
    )

    # init request wrapper
    request_wrapper = GetAccountInfoRequestWrapper(
        header=request_header,
        body=account_query_request
    )
    response_wrapper = account_service.get_account_info(request_wrapper)
    print(response_wrapper)


if __name__ == '__main__':
    main()
